test/index.test.js

```diff
@@ -399,6 +399,47 @@ describe('index', () => {
             });
         });
 
+        describe('Get version from a tag', () => {
+            const config = {
+                name: 'template/test',
+                tag: 'stable'
+            };
+
+            it('throws error when request yields an error', () => {
+                requestMock.rejects(new Error('error'));
+
+                return index.getVersionFromTag(config)
+                    .then(() => assert.fail('should not get here'))
+                    .catch((err) => {
+                        assert.equal(err.message, 'error');
+                    });
+            });
+
+            it('succeeds and does not throw an error if request status code is 200', () => {
+                const responseFake = {
+                    statusCode: 200,
+                    body: templateConfig
+                };
+
+                requestMock.resolves(responseFake);
+
+                return index.getVersionFromTag(config)
+                    .then((result) => {
+                        assert.deepEqual(result, '1.0.0');
+                        assert.calledWith(requestMock, {
+                            method: 'GET',
+                            url,
```

Comment:
The URL here is pointing to the wrong endpoint for this test; `url` is set to hit:
```
/v4/templates/{templateName}/tags/{tagName}
```

You likely want to hit:

```
/v4/templates/{name}/{versionOrTag}
```